File: automation-export/template/samples/node/hello-world.test.js
/**
 * Tests for hello-world module.
 */

const { helloWorld } = require('./hello-world');

describe('helloWorld', () => {
  test('should return default greeting', () => {
    const result = helloWorld();
    expect(result).toBe('Hello, World!');
  });

  test('should return custom name greeting', () => {
    const result = helloWorld('Node.js');
    expect(result).toBe('Hello, Node.js!');
  });

  test('should throw TypeError for non-string name', () => {
    expect(() => helloWorld(123)).toThrow(TypeError);
    expect(() => helloWorld(123)).toThrow('Name must be a string');

    expect(() => helloWorld(null)).toThrow(TypeError);
    expect(() => helloWorld(undefined)).toThrow(TypeError);
  });

  test('should throw Error for empty name', () => {
    expect(() => helloWorld('')).toThrow(Error);
    expect(() => helloWorld('')).toThrow('Name cannot be empty');

    expect(() => helloWorld('   ')).toThrow(Error);
    expect(() => helloWorld('   ')).toThrow('Name cannot be empty');
  });

  describe('Integration Tests', () => {
    test('should handle multiple greetings', () => {
      const names = ['Alice', 'Bob', 'Charlie'];
      const results = names.map(name => helloWorld(name));
      const expected = ['Hello, Alice!', 'Hello, Bob!', 'Hello, Charlie!'];

      expect(results).toEqual(expected);
    });

    test('should handle unicode names', () => {
      const result1 = helloWorld('世界');
      expect(result1).toBe('Hello, 世界!');

      const result2 = helloWorld('José');
      expect(result2).toBe('Hello, José!');
    });

    test('should handle special characters', () => {
      const specialNames = ['@user', '#hashtag', '$money', '%percent'];

      specialNames.forEach(name => {
        const result = helloWorld(name);
        expect(result).toBe(`Hello, ${name}!`);
        expect(result).toMatch(/^Hello, .+!$/);
      });
    });
  });
});

File: automation-export/template/samples/java/src/test/java/com/example/helloworld/HelloWorldTest.java
package com.example.helloworld;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import static org.junit.jupiter.api.Assertions.*;

/**
 * Tests for HelloWorld class.
 */
class HelloWorldTest {

    @Test
    @DisplayName("Should return default greeting")
    void testHelloWorldDefault() {
        String result = HelloWorld.helloWorld();
        assertEquals("Hello, World!", result);
    }

    @Test
    @DisplayName("Should return custom name greeting")
    void testHelloWorldCustomName() {
        String result = HelloWorld.helloWorld("Java");
        assertEquals("Hello, Java!", result);
    }

    @Test
    @DisplayName("Should throw exception for null name")
    void testHelloWorldNullName() {
        IllegalArgumentException exception = assertThrows(
            IllegalArgumentException.class,
            () -> HelloWorld.helloWorld(null)
        );
        assertEquals("Name cannot be null", exception.getMessage());
    }

    @Test
    @DisplayName("Should throw exception for empty name")
    void testHelloWorldEmptyName() {
        IllegalArgumentException exception = assertThrows(
            IllegalArgumentException.class,
            () -> HelloWorld.helloWorld("")
        );
        assertEquals("Name cannot be empty", exception.getMessage());
    }

    @Test
    @DisplayName("Should throw exception for whitespace-only name")
    void testHelloWorldWhitespaceOnlyName() {
        IllegalArgumentException exception = assertThrows(
            IllegalArgumentException.class,
            () -> HelloWorld.helloWorld("   ")
        );
        assertEquals("Name cannot be empty", exception.getMessage());
    }

    @Nested
    @DisplayName("Integration Tests")
    class IntegrationTests {

        @Test
        @DisplayName("Should handle multiple greetings")
        void testMultipleGreetings() {
            String[] names = {"Alice", "Bob", "Charlie"};
            String[] expected = {"Hello, Alice!", "Hello, Bob!", "Hello, Charlie!"};

            for (int i = 0; i < names.length; i++) {
                String result = HelloWorld.helloWorld(names[i]);
                assertEquals(expected[i], result);
            }
        }

        @Test
        @DisplayName("Should handle unicode names")
        void testUnicodeNames() {
            String result1 = HelloWorld.helloWorld("世界");
            assertEquals("Hello, 世界!", result1);

            String result2 = HelloWorld.helloWorld("José");
            assertEquals("Hello, José!", result2);
        }
    }
}
